zerver/lib/email_mirror.py

```diff
@@ -415,6 +416,13 @@ def process_stream_message(to: str, message: EmailMessage) -> None:
     subject_header = message.get("Subject", "")
     subject = strip_from_subject(subject_header) or "(no topic)"
 
+    # We don't want to reject email messages with disallowed characters in the Subject,
+    # so we just make some modifications to the string to make it
+    # a valid Zulip topic name.
+    subject = "".join(
+        [char if is_character_printable(char) else "\N{REPLACEMENT CHARACTER}" for char in subject]
```

Comment:
does this mean `REPLACEMENT CHARACTER` appears in the output string? That seems problematic; we probably just want it to be replaced by the empty string.

Reply:
That's right. I went with that because we settled on that approach when fixing stream names - https://github.com/zulip/zulip/commit/8affb8e4a3744e604b88157a918ef690203cbfa8#diff-0a2fbea2a4b4debe24a0abe93a8768ddd3c26ce7209bda77149667ea1793c43eR40 

But pushed a version of the commit with just removing the invalid characters now and adjusted the test, in case that's better indeed. I can bring back the `"\N{REPLACEMENT CHARACTER}" ` if we want.

Reply:
I think it's probably better to just strip them, given the length limits in topics.